src/connection.ts

```diff
@@ -89,29 +96,31 @@ export class Connection {
   }
 
   static connect(params: ConnectionParams, logger?: Logger): Promise<Connection> {
-    return new Connection(logger ?? new NullLogger()).start(params)
+    return new Connection(logger ?? new NullLogger(), params).start()
   }
 
-  public start(params: ConnectionParams): Promise<Connection> {
-    this.registerListeners(params.listeners)
+  public start(): Promise<Connection> {
+    this.registerListeners(this.params.listeners)
```

Comment:
Would have been easier to simply add a line `const params = this.params`, makes everything else less verbose